apps/wpcom-block-editor/src/calypso/iframe-bridge-server.js

```diff
@@ -611,59 +611,58 @@ function openCustomizer( calypsoPort ) {
  *
  * @param {MessagePort} calypsoPort Port used for communication with parent frame.
  */
-async function openTemplatePartEditor( calypsoPort ) {
+function setupEditTemplateLinks( calypsoPort ) {
 	// We only want this when editing a full site page.
 	if ( ! window.fullSiteEditing || 'page' !== window.fullSiteEditing.editorPostType ) {
 		return;
 	}
 
-	const getTemplatePartLinks = async () =>
-		new Promise( resolve => {
-			const unsubscribe = subscribe( () => {
-				const currentPost = select( 'core/editor' ).getCurrentPost();
-				const initialized = currentPost && currentPost.id;
+	const handleNewTemplateLinks = link => {
+		const templateId = parseInt( getQueryArg( link.getAttribute( 'href' ), 'post' ), 10 );
 
-				if ( ! initialized ) {
-					return;
-				}
-
-				unsubscribe();
-
-				const interval = setInterval( () => {
-					const links = document.querySelectorAll(
-						'[data-type="a8c/template"] .template-block__overlay a'
-					);
-					const blocks = select( 'core/editor' )
-						.getBlocks()
-						.filter( block => block.name === 'a8c/template' );
-
-					if ( links.length !== blocks.length ) {
-						return;
-					}
-
-					clearInterval( interval );
-					resolve( links );
-				} );
-			} );
-		} );
-
-	const editTemplatePartLinks = await getTemplatePartLinks();
+		const { port1, port2 } = new MessageChannel();
 
-	editTemplatePartLinks.forEach( link => {
-		const templatePartId = parseInt( getQueryArg( link.getAttribute( 'href' ), 'post' ), 10 );
+		port1.onmessage = ( { data } ) => {
+			link.setAttribute( 'target', '_parent' );
+			link.setAttribute( 'href', data );
+		};
 
-		const { port1, port2 } = new MessageChannel();
+		// Ask for another URl for the template:
 		calypsoPort.postMessage(
 			{
-				action: 'getTemplatePartEditorUrl',
-				payload: { templatePartId },
+				action: 'getTemplateEditorUrl',
+				payload: { templateId },
 			},
 			[ port2 ]
 		);
-		port1.onmessage = ( { data } ) => {
-			link.setAttribute( 'target', '_parent' );
-			link.setAttribute( 'href', data );
-		};
+	};
+
+	subscribe( () => {
+		const currentPost = select( 'core/editor' ).getCurrentPost();
+		const initialized = currentPost && currentPost.id;
+
+		if ( ! initialized ) {
+			return;
+		}
+
+		// When the template block becomes selected, we want to change
+		// the link of the "edit template" button.
+		const selectedBlock = select( 'core/editor' ).getSelectedBlock();
+
+		if ( selectedBlock && 'a8c/template' === selectedBlock.name ) {
+			const getImpendingLinks = setInterval( () => {
+				const editTemplateLink = document.querySelector(
+					'[data-type="a8c/template"] .template-block__overlay a'
+				);
```

Comment:
Is there not a better way for us to pass this along?

On template, select we wait for the following element to render:
<img width="1312" alt="Screen Shot 2019-08-13 at 5 44 08 PM" src="https://user-images.githubusercontent.com/1270189/62986811-1e52f800-bdf2-11e9-96f7-dd64ce1dde73.png">

Reply:
🤔 I guess we did this previously, so I won't block on this. I'm thinking we can pass along both of these links in a more robust way, as a follow up.